### PR TITLE
SIDE-1090 - better error handling in LDKs

### PR DESCRIPTION
**Pull request by bbrewer-oliveai**
Adds client side, low-level (connection failure, message structure problems, etc.) error logging to each LDK client.

### CODE REVIEW

File: ldk/node/src/hostClients/fileSystemClient.int.test.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import { FileSystemClient } from './fileSystemClient';
```
Comment (bbrewer-oliveai) [inline ldk/node/src/hostClients/fileSystemClient.int.test.ts:1]: will remove

File: ldk/node/src/hostClients/whisperClient.int.test.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import FileSystemClient from './whisperClient';
```
Comment (bbrewer-oliveai) [inline ldk/node/src/hostClients/whisperClient.int.test.ts:1]: will remove

**Review by richseviora — APPROVED** [review]
This looks really good!

Can you add a comment with what these sort of errors look like in the log? I'll probably draw back a bit on the logging in my Filesystem PR

**Comment by bbrewer-oliveai** [comment]
> This looks really good!
> 
> Can you add a comment (in the PR) with what these sort of errors look like in the log? I'll probably draw back a bit on the logging in my Filesystem PR

@richseviora and anyone else
Example of what this comes out as, say for example, if the session is missing from the request - it uses the logger provided by the Loop. The low-level error is also passed on to the consuming Loop as a reject or error part of the streaming callback (NodeJS, for example), and won't crash the loop, so Loop users can still deal with it gracefully if desired. C# is one exception where users will need to catch the exception still.

The only notable things about the log are the error message itself, and then the service and method that failed.
```shell
2021-01-06T14:15:58-05:00 ERR Client exception error="rpc error: code = Unknown desc = proto session is nil pointer" @module=example-network component=loop metadata={"appId":"d9d527bb-7440-490e-80da-201d0666aca2","author":"Ben Brewer","authorId":"5fce6f8c0f91ea0076089380","colorStrip":"#73ff57","created":"2021-01-05T23:52:25.08Z","description":"Description","documentation":"","entrypoint":[],"loopLibraryLogo":"","loopWhisperLogo":"","name":"network","orgId":"Olive","organization":"Olive","semverMac":"1.0.0","semverWindows":"","specification":"1","updated":"2021-01-05T23:52:25.08Z","uploadId":"","version":""} method=HTTPRequest service=Network timestamp=2021-01-06T14:15:58.063-0500
```

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/ExceptionLoggingInterceptor.cs [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+                return action.Result;
+            });
+        }
+
+        private static IAsyncStreamReader<TResponse> WrapStreamHandler<TResponse>(IAsyncStreamReader<TResponse> call, ILogger logger)
+        {
+            return new AsyncStreamReaderWrapper<TResponse>(call, logger);
+        }
+
+        private class MethodExceptionLogger<TRequest, TResponse> : ILogger where TRequest : class where TResponse : class
```
Comment (bbrewer-oliveai) [inline ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/ExceptionLoggingInterceptor.cs:75]: `WithFields` seems like a less bulky alternative to this that I'll do

File: ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/ExceptionLoggingInterceptor.cs [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        private static Task<TResponse> WrapContinueHandler<TResponse>(Task<TResponse> task, ILogger logger)
+        {
+            return task.ContinueWith((action) =>
+            {
+                if (action.Exception == null) return action.Result;
+                action.Exception.Handle((exception) =>
+                {
+                    if (exception is RpcException)
+                    {
+                        logger.Error(exception.Message);
```
Comment (bbrewer-oliveai) [inline ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/ExceptionLoggingInterceptor.cs:61]: I'm going to have this actually log EVERY exception and then see if there is a way to propagate them without losing them inside other continuations, like the LoggedParser, etc.

File: ldk/go/exceptionLoggingInterceptor.go [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+		eli.logError(method, err)
+	}
+
+	return err
+}
+
+func (eli *ExceptionLoggingInterceptor) NewStream(ctx context.Context, desc *grpc.StreamDesc, method string, opts ...grpc.CallOption) (grpc.ClientStream, error) {
+	stream, err := eli.interceptedConn.NewStream(ctx, desc, method, opts...)
+	if err != nil {
+		eli.logError(method, err)
```
Comment (bbrewer-oliveai) [inline ldk/go/exceptionLoggingInterceptor.go:27]: no clue how gofmt was okay with this :/

Comment (BillyMurrayOlive) [reply ldk/go/exceptionLoggingInterceptor.go:27]: What's wrong with it. I must be missing something.

**Review by BillyMurrayOlive — APPROVED** [review]
I only feel like I can be any use for approving the Go piece of this. I approved the PR, but you should really consider it 1/3 of an approval.

File: ldk/go/exceptionLoggingInterceptor.go (hunk 5 above)
Comment (bbrewer-oliveai) [reply ldk/go/exceptionLoggingInterceptor.go:27]: Oh yeah, it's indenting really far, but then I realized GoLang uses tabs. My bad

**Review by BrettMugglinOlive — APPROVED** [review]

**Pull request merged**